File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/OrderDetailProductDiscountMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.OrderDetailProductDiscountDTO;
import com.robinfood.core.entities.OrderDetailProductDiscountEntity;

public final class OrderDetailProductDiscountMappers {

    private OrderDetailProductDiscountMappers() {
        // this constructor is empty because it is a mapper class
    }

    public static OrderDetailProductDiscountDTO toOrderDetailProductDiscountDTO(
            OrderDetailProductDiscountEntity orderDetailProductDiscountEntity
    ) {
        return new OrderDetailProductDiscountDTO(
                orderDetailProductDiscountEntity.getId(),
                orderDetailProductDiscountEntity.getTypeId(),
                orderDetailProductDiscountEntity.getValue()
        );
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/OrderDetailProductDiscountMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.constants.GlobalConstants.DEFAULT_LONG_VALUE
import com.robinfood.core.dtos.OrderDetailProductDiscountDTO
import com.robinfood.core.entities.OrderDetailProductDiscountEntity
import java.math.BigDecimal

fun OrderDetailProductDiscountEntity.toOrderDetailProductDiscountDTO(): OrderDetailProductDiscountDTO? {
    return if (id == null) {
        null
    } else {
        return OrderDetailProductDiscountDTO(
                id = id,
                typeId = typeId ?: DEFAULT_LONG_VALUE,
                value = value ?: BigDecimal.ZERO
        )
    }
}